models/include/cloe/component/gearbox_actuator.hpp

```diff
@@ -0,0 +1,56 @@
+/*
+ * Copyright 2022 Robert Bosch GmbH
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+/**
+ * \file cloe/component/gearbox_actuator.hpp
+ *
+ */
+
+#pragma once
+#ifndef CLOE_COMPONENT_GEARBOX_ACTUATOR_HPP_
+#define CLOE_COMPONENT_GEARBOX_ACTUATOR_HPP_
+
+#include <cloe/component/actuator.hpp>  // for Actuator
+
+namespace cloe {
+
+struct GearboxRequest {
+  /* 
+   * Holds the requested gear selector position. 
+   *
+   * The sign of this field is linked to the mode of the gear
+   * - positive: driving forward (e.g. a value of 3 means to request the third gear in driving forward mode)
+   * - 0: means that the gear lever is requested to be in neutral position
+   * - negative: reverse mode (e.g. a value of -1 means the first gear in reverse mode is requested)
+   * - int max: means that the transmission is in parking position (can be accessed via std::numeric_limits<int>::max())
+   */
+  int gear_selector{0};
```

Comment:
Can we use a sized type for this maybe? Would be easier for portability when communicating the parking position.

Does this interface come from somewhere?

Reply:
what do you mean by a sized type exactly? i only know sized types that can only be positive, though this value here can be negative as well.

this interface is close to OSI definition of gear_selector, but OSI only offers negative, 0, positive, so i extended with int max for the parking position of the gear selector.

Reply:
changed to int8_t